mountpoint-s3-client/tests/auth.rs

```diff
@@ -0,0 +1,206 @@
+    let (bucket, prefix) = get_test_bucket_and_prefix("test_static_provider");
+
+    // Create one object named "hello"
+    let key = format!("{prefix}/hello");
+    let body = b"hello world!";
+    sdk_client
+        .put_object()
+        .bucket(&bucket)
+        .key(&key)
+        .body(ByteStream::from(Bytes::from_static(body)))
+        .send()
+        .await
+        .unwrap();
+
+    // Get some static credentials by just using the SDK's default provider, which we know works
+    let sdk_provider = DefaultCredentialsChain::builder()
+        .region(Region::new(get_test_region()))
+        .build()
+        .await;
+    let credentials = sdk_provider
+        .provide_credentials()
+        .await
+        .expect("can't get static credentials");
+
+    // Build a S3CrtClient that uses a static credentials provider with the creds we just got
+    let config = CredentialsProviderStaticOptions {
+        access_key_id: credentials.access_key_id(),
+        secret_access_key: credentials.secret_access_key(),
+        session_token: credentials.session_token(),
+    };
+    let provider = CredentialsProvider::new_static(&Allocator::default(), config).expect("couldn't create provider");
+    let config = S3ClientConfig {
+        auth_config: S3ClientAuthConfig::Provider(provider),
+        ..Default::default()
+    };
+    let client = S3CrtClient::new(&get_test_region(), config).expect("couldn't create client");
+
+    let result = client
+        .get_object(&bucket, &key, None, None)
+        .await
+        .expect("get_object should succeed");
+    check_get_result(result, None, &body[..]).await;
+
+    // Try it again but corrupt the credentials so that we know they're actually being used. The
+    // client can't tell that the corrupted secret access key is invalid, so the request gets sent,
+    // but fails.
+    let bogus_secret_access_key = credentials.secret_access_key().split_at(5).0;
+    let config = CredentialsProviderStaticOptions {
+        access_key_id: credentials.access_key_id(),
+        secret_access_key: bogus_secret_access_key,
+        session_token: credentials.session_token(),
+    };
+    let provider = CredentialsProvider::new_static(&Allocator::default(), config).expect("couldn't create provider");
+    let config = S3ClientConfig {
+        auth_config: S3ClientAuthConfig::Provider(provider),
+        ..Default::default()
+    };
+    let client = S3CrtClient::new(&get_test_region(), config).expect("couldn't create client");
+
+    let mut request = client
+        .get_object(&bucket, &key, None, None)
+        .await
+        .expect("get_object request should be sent");
+
+    let _error = request
+        .next()
+        .await
+        .unwrap()
+        .expect_err("bogus credentials should not work");
+}
+
+/// Test creating a client with the profile credentials provider
+///
+/// This is complicated because CLI profiles are inherently global state, but we want to isolate the
```

Comment:
Environment variables are global in the sense that they're shared by the entire process, which might have many threads running. Rust runs every test in a separate thread in the same process, so all those threads share the same environment variables. So this test can't just play around with those variables since they'll be used by other threads too.

Reply:
Got it. Thanks